src/azure-cli/azure/cli/command_modules/vm/_template_builder.py

```diff
@@ -304,12 +304,19 @@ def _build_os_profile():
         if enable_auto_update is not None and custom_image_os_type.lower() == 'windows':
             os_profile['windowsConfiguration']['enableAutomaticUpdates'] = enable_auto_update
 
+        # Windows patch settings
```

Comment:
should we add some validation on combination of patch_mode value and os?

Reply:
This is exactly the disadvantage of merging two parameters into one. The validation was done by Azure CLI framework as long as we define `arg_type=get_enum_type(self.get_models('LinuxVMGuestPatchMode')`.

Reply:
Added